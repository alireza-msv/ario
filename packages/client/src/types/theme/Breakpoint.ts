enum Breakpoint {
  SM = 'sm',
  MD = 'md',
  LG = 'lg',
  XL = 'xl',
  XXL = 'xxl',
}

export default Breakpoint;
