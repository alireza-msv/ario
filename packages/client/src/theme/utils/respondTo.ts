import { css, FlattenSimpleInterpolation } from 'styled-components';
import Breakpoint from '~/types/theme/Breakpoint';
import theme from '../theme';

const respondTo = (breakpoint: Breakpoint, content: FlattenSimpleInterpolation): FlattenSimpleInterpolation => css`
  @media(min-width: ${theme.breakpoints[breakpoint]}) {
    ${content}
  }
`;

export default respondTo;
