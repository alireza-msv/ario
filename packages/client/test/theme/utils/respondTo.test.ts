import { css } from 'styled-components';
import respondTo from '~/theme/utils/respondTo';
import Breakpoint from '~/types/theme/Breakpoint';

describe('respondTo tests', () => {
  it('should create mediaQuery for SM breakpoint', () => {
    const wrapper = respondTo(Breakpoint.SM, css`background-color: red;`);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create mediaQuery for MD breakpoint', () => {
    const wrapper = respondTo(Breakpoint.MD, css`background-color: yellow;`);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create mediaQuery for LG breakpoint', () => {
    const wrapper = respondTo(Breakpoint.LG, css`background-color: pink;`);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create mediaQuery for XL breakpoint', () => {
    const wrapper = respondTo(Breakpoint.XL, css`background-color: aqua;`);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create mediaQuery for XXL breakpoint', () => {
    const wrapper = respondTo(Breakpoint.XXL, css`background-color: purple;`);
    expect(wrapper).toMatchSnapshot();
  });
});
