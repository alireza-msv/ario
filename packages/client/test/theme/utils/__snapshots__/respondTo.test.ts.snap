// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`respondTo tests should create mediaQuery for LG breakpoint 1`] = `
Array [
  "
  @media(min-width: ",
  "992px",
  ") {
    ",
  "background-color: pink;",
  "
  }
",
]
`;

exports[`respondTo tests should create mediaQuery for MD breakpoint 1`] = `
Array [
  "
  @media(min-width: ",
  "768px",
  ") {
    ",
  "background-color: yellow;",
  "
  }
",
]
`;

exports[`respondTo tests should create mediaQuery for SM breakpoint 1`] = `
Array [
  "
  @media(min-width: ",
  "576px",
  ") {
    ",
  "background-color: red;",
  "
  }
",
]
`;

exports[`respondTo tests should create mediaQuery for XL breakpoint 1`] = `
Array [
  "
  @media(min-width: ",
  "1200px",
  ") {
    ",
  "background-color: aqua;",
  "
  }
",
]
`;

exports[`respondTo tests should create mediaQuery for XXL breakpoint 1`] = `
Array [
  "
  @media(min-width: ",
  "1600px",
  ") {
    ",
  "background-color: purple;",
  "
  }
",
]
`;
